Making docs better (#29623)
akka/akka: akka-docs/src/main/paradox/stream/operators/ActorFlow/ask.md

The `ask` operator requires

* the actor `ref`,
* a `makeMessage` function to create the message sent to the actor from the incoming element, and the actor ref accepting the actor's reply message and a timeout.

See also:

* @ref[Flow.ask](../Source-or-Flow/ask.md) for the classic actors variant

## Examples

> The `ActorFlow.ask` sends a message to the actor which expects `Asking` messages which contain the actor ref for replies of type `Reply`. When it receives a reply, it emits it  and the `map` extracts the message `String`.

Reviewer: I'm not sure it is very clear what "it" is here. Perhaps:

```suggestion
The `ActorFlow.ask` sends a message to the actor which expects `Asking` messages which contain the actor ref for replies of type `Reply`. The replies are emitted by the `askFlow` and the `map` extracts the message `String`.
```

Reviewer 2: Maybe we can even remove the double-`which` and some pronouns:
```suggestion
The `ActorFlow.ask` sends a message to the actor. The actor expects `Asking` messages which contain the actor ref for replies of type `Reply`. When the actor for replies receives a reply, the `ActorFlow.ask` stream stage emits the reply and the `map` extracts the message `String`.
```

The `ask` operator requires

* the actor `ref`,
* a `makeMessage` function to create the message sent to the actor from the incoming element, and the actor ref accepting the actor's reply message 
* a timeout.

See also:

* @ref[Flow.ask](../Source-or-Flow/ask.md) for the classic actors variant

## Examples

> The `ActorFlow.ask` sends a message to the actor. The actor expects `Asking` messages which contain the actor ref for replies of type `Reply`. When the actor for replies receives a reply, the `ActorFlow.ask` stream stage emits the reply and the `map` extracts the message `String`.

Reviewer: Nice!

Author: Thanks

**emits** when the futures (in submission order) created by the ask pattern internally are completed

> **backpressure** when the number of futures reaches the configured parallelism and the downstream backpressure

Reviewer: 'backpressure' is intended to be a verb here, so it needs to be 'backpressures' in both cases.

**completes** when upstream completes and all futures have been completed and all elements have been emitted

> **fails** when the passed in an actor terminates, or a timeout exceeds in any of the asks performed

Reviewer: 'the passed in actor' is the subject of this sentence, inserting 'an' here makes it invalid.

While changing the passive 'is exceeded' into the active 'exceeds' is probably nicer, the sentence needs to be rearranged further. Perhaps:

```suggestion
**fails** when the passed-in actor terminates, or when any of the `ask`s exceed a timeout
```
